### DISCUSSION TITLE
Interpolate with feature-state

### DISCUSSION THREAD

**Discussion by zmatez** [maplibre/maplibre-gl-js#4965] [General]
Hi, is there a change this could work?
```json
[
  "interpolate",
  [
    "linear"
  ],
  [
    "feature-state",
    "apteki_total"
  ],
  309,
  "#ffffff",
  1687,
  "#1acfe3"
]
```
It works with ['get',...] but I can't use that. Docs say that feature-state works on data-driven paint properties, soo..?
Else, how can I achieve such effect? I use it on fill-color
Thanks

**Comment by HarelM** [comment]
Every spec definition is documented to be data driven supported and/or feature-state supported, consult with the docs to see which feature support what...

**Comment by zmatez** [comment]
Ok nevermind it works but after this I was removing and adding sources again so it all got reset...
